File: easy/python/binarySearch/binarySearchRecursive.py
def findIndex(array, target, left,right):
	if left>right:
		return -1
	mid = (left+right)//2
	if array[mid]==target:
		return mid
	elif array[mid]>target:
		return findIndex(array, target, left, mid-1)
	else:
		return findIndex(array, target, mid+1, right)
	
		
def binarySearch(array, target):
	return findIndex(array, target, 0, len(array)-1)

File: easy/java/BinarySearch/BinarySearchRecursive.java
package BinarySearch;

public class BinarySearchRecursive {
    public static void main(String[] args) {
        System.out.println(binarySearch(new int[] { 0, 1, 21, 33, 45, 45, 61, 71, 72, 73 }, 33));
    }

    public static int binarySearch(int[] array, int target) {
        return findTarget(array, 0, array.length-1, target);
    }

    private static int findTarget(int[] array, int left, int right, int target) {
        int mid = (left+right) / 2;
         if (array[mid] > target) {
            return findTarget(array, left, mid, target);
        }
       if (array[mid] < target) {
            return findTarget(array, mid-1, right, target);
        }
        if (array[mid] == target) {
            return mid;
        }
        return mid>-1?mid:-1;
    }
}